src/power_grid_model_io/converters/tabular_converter.py

```diff
@@ -242,26 +251,33 @@ def _convert_col_def_to_attribute(
         """
         # To avoid mistakes, the attributes in the mapping should exist. There is one extra attribute called
         # 'extra' in which extra information can be captured.
-        if attr not in pgm_data.dtype.names and attr != "extra":
+        if attr not in pgm_data.dtype.names and attr not in ["extra", "filter"]:
```

Comment:
pro tip: use `()` for inline things. it's one of the few cases where the python interpreter can optimize during parsing because tuples `()` are fixed-size. it's minor in this case, but this stuff accumulates multiplicatively so it can add up

```suggestion
        if attr not in pgm_data.dtype.names and attr not in ("extra", "filter"):
```